include/af/image.h

```diff
@@ -677,6 +677,32 @@ AFAPI array ycbcr2rgb(const array& in, const YCCStd standard=AF_YCC_601);
 AFAPI array rgb2ycbcr(const array& in, const YCCStd standard=AF_YCC_601);
 #endif
 
+#if AF_API_VERSION >= 34
+/**
+   C++ Interface for calculating an image moment
+
+   \param[in]  in is the input image
+   \param[moment] is the moment to calculate
+   \return      the value of the moment
+
+   \ingroup image_func_moments
+ */
+template<typename T> T moments(const array& in, const af_moment moment);
```

Comment:
`af_image_moment` (cases where you pass single image and expect single output) and `af_moments` (where you can have multiple images) might be appropriate I think